### DISCUSSION TITLE
Morse Runner 1.81 - Community Edition

### DISCUSSION THREAD

**Discussion by w7sst** [w7sst/MorseRunner#80] [Announcements]
## Summary
This is the second release of the Morse Runner Community Edition. This version contains:
- Add CQ WW Contest
- Bug fixes.
- Menu and splash screen changes

## Join Our Community
You are invited to join our community effort. For more information, please visit the [Morse Runner Community Edition](https://github.com/w7sst/MorseRunner#readme) project.

Our goal is to create program you love to use, so please help by leaving any feedback you would like to share on our [discussions page](https://github.com/w7sst/MorseRunner/discussions/categories/morse-runner-1-80).

Additionally, you can discuss or leave feedback at [MorseRunnerCE on groups.io](https://groups.io/g/MorseRunnerCE). 

You have several ways to join in on our project and we look forward to sharing your ideas. 

## Installation
Uncompress the zip file to any folder and run `MorseRunner.exe`.

## Release Details
This release contains cumulative changes from several authors.

### Version 1.81 (Oct 2022)
  - Fix bad CWOPS CWT Exchange message (Coded by W7SST)
  - Added CQ WW Contest (Coded by W7SST)
  - Update README.md (by F6FVY)
  - Updated GUI and menu items (Coded by W7SST)

### Version 1.80 (Oct 2022)
  - Added multi-contest support (Coded by W7SST)
  - Added ARRL Field Day contest (Coded by W7SST)
  - Added NCL NAQP contest (Coded by W7SST)

## What's Changed
* Fix #44 - fix bad CWOPS CWT Exchange message when starting without .i… by @w7sst in https://github.com/w7sst/MorseRunner/pull/47
* Quarterly project Roadmap update by @w7sst in https://github.com/w7sst/MorseRunner/pull/50
* Fixes #51 - Add additional 'Adding a contest' comments. by @w7sst in https://github.com/w7sst/MorseRunner/pull/53
* Add CQ WW Contest by @w7sst in https://github.com/w7sst/MorseRunner/pull/54
* Issue #61 - change title bar to 'Morse Runner - Community Edition' by @w7sst in https://github.com/w7sst/MorseRunner/pull/69
* Fix #59 - The 'Help > Web Page' now directs to the GitHub home page by @w7sst in https://github.com/w7sst/MorseRunner/pull/70
* rename Help menu's 'Web Page...' to 'Community Edition Home Page' by @w7sst in https://github.com/w7sst/MorseRunner/pull/71
* Update README.md by @f6fvy in https://github.com/w7sst/MorseRunner/pull/72
* Issue #56 - Update splash screen with latest copyright info (Coded by W7SST) by @w7sst in https://github.com/w7sst/MorseRunner/pull/74
* Issue #60 - Restore missing Help About dialog (Coded by W7SST) by @w7sst in https://github.com/w7sst/MorseRunner/pull/73

## New Contributors
* @f6fvy made their first contribution in https://github.com/w7sst/MorseRunner/pull/72

**Full Changelog**: https://github.com/w7sst/MorseRunner/compare/v1.80...v1.81

<hr /><em>This discussion was created from the release <a href='https://github.com/w7sst/MorseRunner/releases/tag/v1.81'>Morse Runner 1.81 - Community Edition</a>.</em>

**Comment by scotthibbs** [comment]
I asked Mark Holland, KD0EE who has got certificates in contesting to look at CE 1.81. The following was his email to me:

"Hey Scott,
Love, Love , LOVE the program! Been having great fun. Almost as much fun as being on the air!
Only negative is that the key assignments I use on N1MM+ are different from Morse Runner and the program does not let you reassign the keys. But I’m getting used to it.
Thanks for the link to the program."

Assignable keys are a good Idea. Aren't we working on getting N1MM+ compatibility or is it something different?

**Comment by w7sst** [comment]
I agree with the idea of key assignments that match N1MM+, or perhaps key assignments. The later is better to allow compatibility with other logging programs. Please feel free to add a feature request for key assignments.

Second, we are working on N1MM integration. I have not heard back from Steve, @n2ic, to see how he is doing with his testing with N1MM integration. I have given him so code to try. He has it compiling and starting to run. More work is needed after he makes a few changes to N1MM itself. Once N1MM integration is done, then the user can run everything from within N1MM.

**Comment by f6fvy** [comment]
You can allow the user to use other assignments but PLEASE don't modify the default assignments. They are used since decades. Maybe predefined "profiles" would be a good idea as well to match several different contest loggers,

I do have certificates in contesting too ;-)

**Reply by scotthibbs** [reply]
That is the idea here. Have the ability to use the default assignments or change them to your liking. These can be changed individually or have a dropdown of preselects in the radio section to change them as a group with options like "Default Radio", "N1MM Radio", etc.

**Comment by scotthibbs** [comment]
More feedback from Mark Holland, KD0EE. He mentioned three things: 

1. He states in many cases a TU is not sent in a contest. And that at the end of a contact if he doesn't send TU fast enough, (ghosted) he doesn't get credit for the contact even though all the info entered is correct. Should the log check for the last valid call? We both discussed why it wouldn't be good for someone to change the log once they know the call/exchange was bad but he felt it would have been credited a good call if the info was right in real life even if the sender already qsy. 
2. He mentioned that in N1MM he has partial checking from his previous logs in the contest, he states some of these files are downloaded from N1MM before the contest. We discussed if this was a logging software feature or if it should be in MorseRunner since it is a pile up simulator. He's excited about the possibility to add his own previous contest logs for the contest list calls and excited about using it with N1MM for the partial call feature.  
3. He mentioned lack of farnsworth for bad calls and wondered if we considered it. I explained how we decided it is not a Morse learning program and why it wasn't likely to have it. Then he admitted it is usually him having to slow down his "E E" when people copy an "I" instead. I mentioned again this is a pile up simulator but it does happen during a contest so I'm passing it along.  

At this point, I realized I might be swatting down his ideas and said no idea is a bad one and to keep giving me feedback because he is helping us change the program for the better. For example, his last feedback gave us the idea of assignable keys and how this would help us make this compatible with other loggers and software. I explained his contribution of feedback is cherished and used. All in all, he loves it. The fact that I shared 1.81 with him a week ago and he's still using it to have feedback is phenomenal. He's excited to have the ARRL DX as the next contest in 1.82. 

Let me know if an issue needs to be made for any of this. -Scott

**Comment by f6fvy** [comment]
Concerning the partials and N+1 windows and others, I think that it must remain reserved to the logging program. Especially because each program has its own specificities and the operator has to get used to it visually as if he/she was in real contest. MRC should act almost as a background application or a "service".
